drivers/storage/portworx/component/securitycontextconstraints.go

```diff
@@ -81,6 +81,35 @@ func (s *scc) crdExists(cluster *opcorev1.StorageCluster) bool {
 	}
 }
 
+func deepEqualScc(scc, existingScc *ocp_secv1.SecurityContextConstraints) bool {
+	if !(scc.Priority == nil && existingScc.Priority == nil ||
+		scc.Priority != nil && existingScc.Priority != nil && *scc.Priority == *existingScc.Priority) ||
+		!(scc.AllowPrivilegeEscalation == nil && existingScc.AllowPrivilegeEscalation == nil ||
+			scc.AllowPrivilegeEscalation != nil && existingScc.AllowPrivilegeEscalation != nil &&
+				*scc.AllowPrivilegeEscalation == *existingScc.AllowPrivilegeEscalation) {
+		return false
+	}
+
+	// Skip comparing auto-generated fields
+	sccCopy := scc.DeepCopy()
+	sccCopy.ResourceVersion = existingScc.ResourceVersion
+	sccCopy.ObjectMeta.UID = existingScc.ObjectMeta.UID
+	sccCopy.ObjectMeta.Generation = existingScc.ObjectMeta.Generation
+	sccCopy.ObjectMeta.CreationTimestamp = existingScc.ObjectMeta.CreationTimestamp
+	sccCopy.ObjectMeta.ManagedFields = existingScc.ObjectMeta.ManagedFields
+	sccCopy.ObjectMeta.GenerateName = existingScc.ObjectMeta.GenerateName
+	sccCopy.ObjectMeta.Finalizers = existingScc.ObjectMeta.Finalizers
+	sccCopy.ObjectMeta.OwnerReferences = existingScc.ObjectMeta.OwnerReferences
+	sccCopy.TypeMeta.Kind = existingScc.TypeMeta.Kind
+	sccCopy.TypeMeta.APIVersion = existingScc.TypeMeta.APIVersion
+	sccCopy.Priority = existingScc.Priority
+	sccCopy.AllowPrivilegeEscalation = existingScc.AllowPrivilegeEscalation
+	if scc.Groups == nil {
+		sccCopy.Groups = make([]string, 0)
```

Comment:
a little bit confused, what was the change here before? the `Groups` field can be nil from existing scc, and deep equal comparing empty slice and nil will be false.

Reply:
scc can be nil, but existing scc is auto converted to empty slice, so I set sccCopy's value to empty slice.